libsolidity/codegen/ir/IRGeneratorForStatements.h

```diff
@@ -191,6 +192,8 @@ class IRGeneratorForStatements: public ASTConstVisitor
 	static Type const& type(Expression const& _expression);
 
 	void setLocation(ASTNode const& _node);
+	std::ostringstream& emitCode();
```

Comment:
Can we somehow protect `m_code` from being accessed directly? For example by making it a private member of a base class that only contains `m_code` and `emitCode()` (which is protected)?